x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/BlockUtils.java

```diff
@@ -139,9 +139,14 @@ public static Block[] fromList(List<List<Object>> list) {
     }
 
     public static Block deepCopyOf(Block block) {
-        Block.Builder builder = block.elementType().newBlockBuilder(block.getPositionCount());
-        builder.copyFrom(block, 0, block.getPositionCount());
-        return builder.build();
+        return deepCopyOf(block, BlockFactory.getNonBreakingInstance());
```

Comment:
Should we use the `block.blockFactory()` here? Or are we trying not to use that one.

Reply:
let's leave this for now. There is only one usage, and we'll quash it soon enough.